CONTRIBUTING.md

```diff
@@ -1,3 +1,37 @@
 # Contributing
 
-Newly contributed code must be licensed under [GNU GPL 2.0](https://www.gnu.org/licenses/gpl-2.0.html) with “or any later version” clause or a [compatible license](https://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses).
+## Code
+Contributing code is very easy to do once you get the hang of it. We use GitHub for our online repository and issue tracker, so you'll need an account first. After you've done that and are comfortable with how the platform and git itself works, read on.
+
+### Set Up Your Repository
+1. Fork [LibreSprite](https://github.com/LibreSprite/LibreSprite) so you have a personal repository to push to.
+2. Clone your personal repository with `git clone --recursive https://github.com/YOUR-USERNAME/LibreSprite`. Remember to change the URL to your repository's. The `--recursive` is used to download the third party libraries for building.
+3. Move into the directory, on Linux this is done with `cd LibreSprite`.
+3. Follow the [instructions](INSTALL.md) to create a build and make sure your local copy is working.
+4. Add an upstream remote so you can get other developer's updates with `git remote add upstream https://github.com/LibreSprite/LibreSprite`.
+
+### Updating Your Repository
+These instructions will update both your local repository and online fork.
+1. `git fetch upstream` Will download any new changes from the official LibreSprite repository.
```

Comment:
Shouldn't this be `git pull upstream`? AFAIK `fetch` only checks for changes, but doesn't actually update the local git repo.

Reply:
This is partially true, it checks and downloads all updates but doesn't update master. The guide is written plainly so it doesn't assume the reader will be on master at the time. Personally, breaking the task into explicit instructions (pull is an implicit fetch/merge) I find to be more appealing. You could also checkout master and pull upstream, either way works.